### PR TITLE
db, state: rename polkadb and migrate db types

### LINKED ISSUE #545: Refactor polkadb
- [ ] Move blockDB and stateDB to state package
- [ ] Rename polkadb to db

### PR DESCRIPTION
**Pull request by ryanchristo**
## Changes

This pull request renames `polkadb` to `db` and moves types `BlockDB` and `StateDB` to `state`.

## Tests:
```
go test ./... --short
```

### Issues:

closes #545

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Our packages:
 | `dot` | wraps other packages to allow a complete client |
 | `internal` | internal api  |
 | `p2p` | peer-to-peer service using libp2p |
-| `polkadb` | database implemenation using badgerDB |
+| `db` | database implemenation using badgerDB |
```
Comment (noot) [inline README.md]: move above dot to keep this alphabetical?

File: consensus/babe/babe_test.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ import (
 	"github.com/ChainSafe/gossamer/core/blocktree"
 	"github.com/ChainSafe/gossamer/core/types"
 	"github.com/ChainSafe/gossamer/crypto/sr25519"
-	db "github.com/ChainSafe/gossamer/polkadb"
+	db "github.com/ChainSafe/gossamer/db"
```
Comment (noot) [inline consensus/babe/babe_test.go]: don't need the package alias anymore

File: core/blocktree/blocktree_test.go [hunk 3]
```diff
@@ -20,10 +20,10 @@ import (
 	"math/big"
 	"testing"
 
-	"github.com/ChainSafe/gossamer/core/types"
-
 	"github.com/ChainSafe/gossamer/common"
-	db "github.com/ChainSafe/gossamer/polkadb"
+	"github.com/ChainSafe/gossamer/core/types"
+	db "github.com/ChainSafe/gossamer/db"
```
Comment (noot) [inline core/blocktree/blocktree_test.go]: same here

File: trie/database_test.go [hunk 4]
```diff
@@ -25,7 +25,7 @@ import (
 	"testing"
 
 	"github.com/ChainSafe/gossamer/config/genesis"
-	db "github.com/ChainSafe/gossamer/polkadb"
+	db "github.com/ChainSafe/gossamer/db"
```
Comment (noot) [inline trie/database_test.go]: same here

File: state/block.go [hunk 5]
```diff
@@ -9,27 +9,41 @@ import (
...
 )
 
+// BlockDB stores block's in an underlying Database
+type BlockDB struct {
+	Db db.Database
+}
+
 // blockState defines fields for manipulating the state of blocks, such as BlockTree, BlockDB and Header
 type blockState struct {
-	bt           *blocktree.BlockTree
```
[the remark this answers is not in the document]
Comment (ryanchristo) [reply state/block.go]: Ah! Good catch. I was getting an import cycle error and forgot to clean up. All good now.
https://github.com/ChainSafe/gossamer/issues/545#issuecomment-580422908

**Pull request merged**